include/chopper/layout/ibf_query_cost.hpp

```diff
@@ -20,34 +20,44 @@ class ibf_query_cost
     ibf_query_cost & operator=(ibf_query_cost &&) = default;
     ~ibf_query_cost() = default;
 
-    constexpr static double exact(size_t const t_max)
+    static double exact(double const fpr, size_t const t_max)
     {
+        auto it = cost_factor.find(fpr);
+
+        if (it == cost_factor.end())
+            it = find_closest_fpr(fpr);
+
         if (contains(t_max))
-            return cost_factor[position(t_max)];
+            return it->second[position(t_max)];
         else
             throw std::invalid_argument("No exact data available for this t_max.");
     }
 
-    constexpr static double interpolated(size_t const t_max)
+    static double interpolated(double const fpr, size_t const t_max)
```

Comment:
I feel like `t_max` should be first, because `t_max` seems more important.
This is just a subjective opinion on what I would expect to be the first argument when writing `ibf_query_cost::interpolated(...`

A practical use would be that you can write
```cpp
static double interpolated(size_t const t_max, double const fpr = 0.0125)
```
and wouldn't have to change the tests in this PR :D

Reply:
I thought you look up fpr first in the map and then the tmax. But not changing the tests is nice so I'll change it

Reply:
hmm but now that I think about it, having a default fpr doesn't make sense at all. You would wonder why 0.0125 is better then the others.

Reply:
I thought you will do a follow-up where you enter the correct values?
Then it would be ok to set a default now, and remove the default once the values actually differ

Reply:
But now the PR is ready as it is :D